qemu/tests/hugepage_specify_node.py

```diff
@@ -61,26 +63,36 @@ def run(test, params, env):
             params["target_num_node%s" % idle_node] = math.ceil(idle_node_mem / hugepage_size)
             error_context.context("Setup huge pages for idle node%s." %
                                   idle_node, test.log.info)
-        params["setup_hugepages"] = "yes"
         hp_config = test_setup.HugePageConfig(params)
-        hp_config.setup()
-        params["qemu_command_prefix"] = "numactl --membind=%s" % node_id
-        params["start_vm"] = "yes"
-        params["hugepage_path"] = hp_config.hugepage_path
-        env_process.preprocess_vm(test, params, env, params["main_vm"])
         try:
+            hp_config.setup()
+            params["qemu_command_prefix"] = "numactl --membind=%s" % node_id
+            params["start_vm"] = "yes"
+            params["hugepage_path"] = hp_config.hugepage_path
+            env_process.preprocess_vm(test, params, env, params["main_vm"])
             vm = env.get_vm(params["main_vm"])
-            vm.verify_alive()
-            vm.wait_for_login()
+            try:
+                vm.verify_alive()
+                vm.wait_for_login()
 
-            meminfo = host_numa_node.get_all_node_meminfo()
-            for index in check_list:
-                error_context.base_context("Check process HugePages Free on host "
-                                           "numa node %s." % index, test.log.info)
-                hugepages_free = int(meminfo[index]["HugePages_Free"])
-                if int(node_meminfo[index]["HugePages_Free"]) > hugepages_free:
-                    test.fail("Qemu still use HugePages from other node."
-                              "Expect: node%s, used: node%s." % (node_id, index))
+                meminfo = host_numa_node.get_all_node_meminfo()
+                for index in check_list:
+                    error_context.base_context("Check process HugePages Free on host "
+                                               "numa node %s." % index, test.log.info)
+                    hugepages_free = int(meminfo[index]["HugePages_Free"])
+                    if int(node_meminfo[index]["HugePages_Free"]) > hugepages_free:
+                        test.fail("Qemu still use HugePages from other node."
+                                  "Expect: node%s, used: node%s." % (node_id, index))
+            finally:
+                vm.destroy()
+        except Exception as details:
+            # Check if there have enough contiguous Memory on the host node
+            details = str(details)
+            if re.findall(r"Cannot set %s hugepages" % huge_pages_num, details):
+                test.cancel("Host node does not have enough contiguous memory "
+                            "to run the test, skipping test...")
+            test.error(details)
         finally:
-            vm.destroy()
             hp_config.cleanup()
+            while (huge_pages_num == 0):
```

Comment:
Please use the `wait_for` since it will be a death loop if there is a bug that huge_pages_num never equals 0.

Reply:
has been corrected